packages/apply/index.ts

```diff
@@ -0,0 +1,396 @@
+      if (version) pkg.pkg.package.version = version;
+    }
+  }
+  return pkg;
+};
+
+const bumpDeps = ({
+  packageFile,
+  dep,
+  bumpType,
+}: {
+  packageFile: PackageFile;
+  dep: string;
+  bumpType: string;
+}) => {
+  let pkg = { ...packageFile };
+
+  if (pkg.pkg && pkg.vfile && pkg.pkg.dependencies)
+    Object.keys(pkg.pkg.dependencies).forEach((existingDep) => {
+      if (
+        existingDep === dep &&
+        pkg.pkg &&
+        pkg.pkg!.dependencies &&
+        pkg.pkg!.dependencies[dep]
+      ) {
+        if (pkg.vfile!.extname === ".json") {
+          // for javascript
+          let version = semver.inc(
+            pkg.pkg.dependencies[dep],
+            // @ts-ignore TODO deal with ReleaseType
+            bumpType
+          );
+          if (version) pkg.pkg.dependencies[dep] = version;
+        } else if (pkg.vfile!.extname === ".toml") {
+          // for rust
+          if (typeof pkg.pkg.dependencies[dep] === "object") {
+            // @ts-ignore TODO deal with nest toml
+            pkg.pkg.dependencies[dep].version = incWithPartials(
+              // @ts-ignore TODO deal with nest toml
+              pkg.pkg.dependencies[dep].version,
+              bumpType
+            );
+          } else {
+            let version = incWithPartials(pkg.pkg.dependencies[dep], bumpType);
+            if (version) pkg.pkg.dependencies[dep] = version;
+          }
+        }
+      }
+    });
+
+  if (pkg.pkg && pkg.vfile && pkg.pkg.devDependencies)
+    Object.keys(pkg.pkg.devDependencies).forEach((existingDep) => {
+      if (
+        existingDep === dep &&
+        pkg.pkg &&
+        pkg.pkg!.devDependencies &&
+        pkg.pkg!.devDependencies[dep]
+      ) {
+        if (pkg.vfile!.extname === ".json") {
+          // for javascript
+          // @ts-ignore TODO deal with ReleaseType
+          let version = semver.inc(pkg.pkg.devDependencies[dep], bumpType);
+          if (version) pkg.pkg.devDependencies[dep] = version;
+        }
+      }
+    });
+
+  return pkg;
+};
+
+const incWithPartials = (version: string, bumpType: string) => {
+  if (semver.valid(version)) {
+    // @ts-ignore TODO deal with ReleaseType
+    return semver.inc(version, bumpType);
+  } else {
+    try {
+      const coerced = semver.coerce(version);
+      if (!coerced)
+        throw new Error(
+          `Cannot bump ${version} with ${bumpType}. Is it a valid version number?`
+        );
+      // @ts-ignore TODO deal with ReleaseType
+      const fullVersion = semver.inc(coerced, bumpType).split(".");
+      if (version.split(".").length === 2) {
+        return [fullVersion[0], fullVersion[1]].join(".");
+      } else if (version.split(".").length === 1) {
+        return fullVersion[0];
+      } else {
+        // failsafe is better than null
+        return fullVersion.join(".");
+      }
+    } catch {
+      // failsafe is better than null
+      // @ts-ignore TODO what should we _really_ do here?
```

Comment:
i think this should throw as well. However, I can say that there have been a number of times in my use of covector, that the error message wasn't always very helpful. Maybe a complete stack trace would be helpful - or at least all variable values...

Reply:
True, we really don't want to proceed if we get to that point. This whole block could use some attention; it bleeds into the other comment chain.